bgpd/bgp_updgrp_packet.c

```diff
@@ -763,7 +763,13 @@ subgroup_update_packet (struct update_subgroup *subgrp)
 
 	  if (rn->prn)
 	    prd = (struct prefix_rd *) &rn->prn->p;
-          tag = bgp_adv_label(rn, binfo, peer, afi, safi);
+
+          if (safi == SAFI_LABELED_UNICAST)
+            tag = bgp_adv_label(rn, binfo, peer, afi, safi);
```

Comment:
Does  bgp_adv_label only make sense for safi == SAFI_LABELED_UNICAST?  If so, shouldn't this logic be moved into bgp_adv_label?